protocol/git/README.md

```diff
@@ -0,0 +1,88 @@
+# Git Protocol
+
+A guide for programming within version control.
+
+Maintain a Repo
+---------------
+
+* Use [GitFlow] as a branch management strategy.
+* Avoid including files in source control that are specific to your
+  development machine or process.
+* Delete local and remote feature branches after merging.
+* Perform work in a feature branch.
+* Rebase frequently to incorporate upstream changes.
+* Use a [pull request] for code reviews.
+
+[GitFlow]: https://nvie.com/posts/a-successful-git-branching-model/
+[pull request]: https://help.github.com/articles/using-pull-requests/
+
+Write a Feature
+---------------
+
+Create a local feature branch based off `develop`.
+
+	
+*Branch Names*
+
+Branch naming style varies slightly depending on the ticketing system we're using:
+
+*Jira*: `feature/<ticket id>-human-readable-reminder, e.g. `feature/HG-123-update-readme`.
+
+*Trello*: `feature/<issue ID from link>`, e.g. `feature/51-share-handsome-pov-via-social-media`.
+
+*Other*: `feature/<short description>`, e.g. `feature/remove-obsolete-links`.
+
+    git checkout develop
+    git pull
+    git checkout -b <branch-name>
+
+Merge frequently to incorporate upstream changes.
+
+    git fetch origin
+    git merge origin/develop
+
+Resolve conflicts. When feature is complete and tests pass, stage the changes.
+
+    git add --all
+
+When you've staged the changes, commit them.
+
+    git status
+    git commit --verbose
+
+Write a [good commit message]. Example format:
+
+    Present-tense summary under 50 characters
+
+    * More information about commit (under 72 characters).
+    * More information about commit (under 72 characters).
+
+    http://project.management-system.com/ticket/123
+
+When the feature is complete, share your branch.
+
+    git push origin <branch-name>
+
+Submit a [GitHub pull request].
+
+Assign reviewers. Ask for a code review in the project's chat room.
+
+[good commit message]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
```

Comment:
I really like that article. I also like this one: https://chris.beams.io/posts/git-commit/  – to me personally feels a little easier to comprehend and follow. 

Additionally, what do you think about the "Tips fo Commit Messages" from here – https://github.com/handsomecode/iOS-Guidelines/blob/master/VersionControl.md ? I am certainly in favor of having a very consistent way of writing commit messages. Things like "Use the imperative mood in the subject line" and "Use the body to explain what and why vs. how" are important when you're glancing through the commit history.

Reply:
Agreed; the Chris Beams article is stronger, though it's more lengthy. As a bonus, it seems to include the content of the Tips for Commit Messages (which I strongly suspect were taken from that article). Updating to point to the CB article -- thanks for that!

My inclination is to keep this as concise as possible, and not to duplicate the content that we're linking to here; again, happy to reconsider if we feel there's more value in bringing it in to this doc directly.